pkg/server/server.go

```diff
@@ -202,34 +202,38 @@ func createMetrics() metrics {
 	return m
 }
 
-func BuildMapperChain(cfg config.Config) []mapper.Mapper {
+func BuildMapperChain(cfg config.Config) ([]mapper.Mapper, error) {
 	modes := cfg.BackendMode
 	mappers := []mapper.Mapper{}
 	for _, mode := range modes {
 		switch mode {
 		case mapper.ModeFile:
+			fallthrough
+		case mapper.ModeMountedConfigMap:
 			fileMapper, err := file.NewFileMapper(cfg)
 			if err != nil {
-				logrus.Fatalf("backend-mode %q creation failed: %v", mode, err)
+				return nil, fmt.Errorf("backend-mode %q creation failed: %v", mode, err)
```

Comment:
Would server still be running after this Failure ?

Reply:
Nope, caller will panic.